src/main/scala/tilelink/Monitor.scala

```diff
@@ -66,6 +66,18 @@ class TLMonitor(args: TLMonitorArgs, monitorDir: MonitorDirection = MonitorDirec
       c.visibility.map(_.contains(address)).reduce(_ || _)
     }.reduce(_ && _)
 
+  def checkFormatOpcode(opcode: UInt, edge: TLEdge, format: String): Unit = {
+    // read params and check the opcode
+    if (!edge.manager.anySupportAcquireT) monAssert (opcode =/= TLMessages.AcquirePerm, s"'${format} channel should not see AcquirePerm")
+    if (!edge.manager.anySupportAcquireB) monAssert (opcode =/= TLMessages.AcquireBlock, s"'${format} channel should not see AcquireBlock")
```

Comment:
unfortunately, Trunk/Branch is orthogonal to Perm/Block. The former are specified in the `param` field, the latter are specified in the `opcode` field. I don't think we have any diplomatic `edge.manager.anySupport` helper methods for distinguishing between AcquirePerm vs AcquireBlock opcode types, so you'll have to assume that if one is allowed the other is too.

@terpstra @brrmorre should be making the `support` methods more granular in the `v2` versions?

Reply:
Ok, I kind of expected this to fail b/c I have no idea what `anySupportAcquireT` and `anySupportAcquireB` are supposed to do. I was loosely pattern-matching based off of name and clearly guessed wrong.

My goal with this PR is to look at all the params of a node, and write relevant assertions that don't exist yet.

So I think what I need to figure out is exactly what `anySupportAcquireT` and `anySupportAcquireB` mean (please help me if you know/know where I should look for this info).